packages/tc-ui/src/primitives/text/server.jsx

```diff
@@ -9,35 +9,54 @@ const localOnChange = (event, onChange) => {
 };
 
 const EditText = ({
+	isEdit,
 	propertyName,
 	value,
 	required,
 	lockedBy,
 	disabled,
 	isNested,
 	parentCode,
+	type,
 	onChange,
 }) => {
 	const name = !isNested
 		? `${propertyName}${lockedBy || disabled ? '-disabled' : ''}`
 		: '';
 
+	// Special case for locking codes during edit
+	let additionalDescriptiveText;
+	if (type === 'Code' && isEdit) {
```

Comment:
Can you go with `propertyName === 'code'` as the check. `type: 'Code'` used to power a few things, but I'm not sure it does any more and I'm not sure it's worth keeping around really.